programs/drift_vaults/src/state/vault.rs

```diff
@@ -82,38 +82,37 @@ impl Size for Vault {
 const_assert_eq!(Vault::SIZE, std::mem::size_of::<Vault>() + 8);
 
 impl Vault {
-    pub fn apply_management_fee(&mut self, vault_equity: u64, now: i64) -> Result<(u64, u64)> {
+    pub fn apply_management_fee(&mut self, vault_equity: u64, now: i64) -> Result<(i64, i64)> {
         let depositor_equity =
             depositor_shares_to_vault_amount(self.user_shares, self.total_shares, vault_equity)?
-                .cast::<u128>()?;
-        let mut management_fee_payment: u128 = 0;
-        let mut management_fee_shares: u128 = 0;
+                .cast::<i128>()?;
+        let mut management_fee_payment: i128 = 0;
+        let mut management_fee_shares: i128 = 0;
 
-        if self.management_fee > 0 && depositor_equity > 0 {
+        if self.management_fee != 0 && depositor_equity > 0 {
             let since_last = now.safe_sub(self.last_fee_update_ts)?;
 
             management_fee_payment = depositor_equity
                 .safe_mul(self.management_fee.cast()?)?
-                .safe_div(PERCENTAGE_PRECISION)?
+                .safe_div(PERCENTAGE_PRECISION_I128)?
                 .safe_mul(since_last.cast()?)?
-                .safe_div(ONE_YEAR)?
+                .safe_div(ONE_YEAR.cast()?)?
                 .min(depositor_equity.saturating_sub(1));
 
             let new_total_shares_factor: u128 = depositor_equity
-                .cast::<u128>()?
-                .safe_mul(PERCENTAGE_PRECISION)?
-                .safe_div(
-                    depositor_equity
-                        .cast::<u128>()?
-                        .safe_sub(management_fee_payment)?,
-                )?;
+                .safe_mul(PERCENTAGE_PRECISION_I128)?
+                .safe_div(depositor_equity.safe_sub(management_fee_payment)?)?
+                .cast()?;
 
             let new_total_shares = self
                 .total_shares
                 .safe_mul(new_total_shares_factor.cast()?)?
-                .safe_div(PERCENTAGE_PRECISION)?;
+                .safe_div(PERCENTAGE_PRECISION)?
+                .max(self.user_shares);
```

Comment:
so if the manager has 0 shares, they just wouldn't pay in that case?

Reply:
yes, same case for vault depositor who has no shares